### PR TITLE
different methods are now available for setting bestwindow threshold. COMPLETE FOLDER WAS REFORMATED!

### PR DESCRIPTION
**Pull request by jfsehuanes**
Ignore the reformat, this was basically for making things "einheitlich". important changes were performed in bestwindow.py and peakdetection.py. For setting the peakdetection threshold there are now three different approaches available: std, percentile and min-max. These are now included into peakdetection.py

peace out!

### CODE REVIEW

File: thunderfish/Auxiliary.py [hunk 1]
```diff
@@ -5,11 +5,11 @@
 import wave
 import matplotlib.pyplot as plt
 import sys
-#import seaborn as sns
+# import seaborn as sns
```
Comment (janscience) [inline thunderfish/Auxiliary.py]: remove seaborn! Everywhere!

File: thunderfish/bestwindow.py [hunk 2]
```diff
@@ -113,12 +113,11 @@ def clip_args(cfg, rate):
...
     a = cfg.map({'min_fac': 'minClipFactor', 'nbins': 'clipBins'})
-    a['win_indices'] = int(cfg.value('clipWindow')*rate)
+    a['win_indices'] = int(cfg.value('clipWindow') * rate)
     return a
 
 
-def best_window_indices(data, samplerate, single=True, win_size=8., win_shift=0.1, percentile_th=99,
-                        th_factor=0.8, min_clip=-np.inf, max_clip=np.inf,
+def best_window_indices(data, samplerate, single=True, win_size=8., win_shift=0.1, min_clip=-np.inf, max_clip=np.inf,
                         w_cv_interv=1.0, w_ampl=1.0, w_cv_ampl=1.0, tolerance=0.5,
```
Comment (janscience) [inline thunderfish/bestwindow.py:122]: No! You need percentile_th=1 and  th_factor=0.8 as parameter for the function

File: thunderfish/bestwindow.py [hunk 3]
```diff
@@ -194,25 +191,19 @@ def best_window_indices(data, samplerate, single=True, win_size=8., win_shift=0.
...
-    # threshold for peak detection:
-    threshold = np.zeros(len(data))
-    win_shift_indices = int(win_shift * samplerate)
-
-    for inx0 in xrange(0, len(data) - win_shift_indices / 2, win_shift_indices):
-        inx1 = inx0 + win_shift_indices
-        threshold[inx0:inx1] = np.diff(np.percentile(data[inx0:inx1],
-                                                     [percentile_th, 100. - percentile_th])) * th_factor
+    # threshold for peak detection: (can choose between std, percentile and min-max... see peakdetection module)
+    threshold = pkd.percentile_threshold(data, samplerate, win_shift)
```
Comment (janscience) [inline thunderfish/bestwindow.py]: this function should get the percentile_th and th_factor as parameter!

File: thunderfish/peakdetection.py [hunk 4]
```diff
@@ -2,6 +2,84 @@
 import numpy as np
 
 
+def std_threshold(data, samplerate, win_shift, std_factor=5.):
```
Comment (tillraab) [inline thunderfish/peakdetection.py]: maybe rename "std_factor" to "th_factor" to make it consistent for each of the 3 functions ??

File: thunderfish/peakdetection.py [hunk 5]
```diff
@@ -2,6 +2,84 @@
...
+    win_shift_indices = int(win_shift * samplerate)
+
+    for inx0 in xrange(0, len(data) - win_shift_indices / 2, win_shift_indices):
+        inx1 = inx0 + win_shift_indices
+
+        window_min = np.min(data[inx0:inx1])
+        window_max = np.max(data[inx0:inx1])
+
+        threshold[inx0:inx1] = np.squeeze(np.abs(np.diff([window_min, window_max]))) * th_factor
+
```
Comment (janscience) [inline thunderfish/peakdetection.py]: Well,this is simply
(window_max - window_min) \* th_factor

File: thunderfish/peakdetection.py [hunk 6]
```diff
@@ -2,6 +2,84 @@
...
+    :return: threshold: (1-D array). Array with same size as data, including a threshold for each window.
+    """
+    threshold = np.zeros(len(data))
+    win_shift_indices = int(win_shift * samplerate)
+
+    # Make sure percentile_th is correct, regardless of what the user sets as percentile_th
+    if percentile_th < 50.0:
+        percentile_th = 100.0 - percentile_th
+
+    for inx0 in xrange(0, len(data) - win_shift_indices / 2, win_shift_indices):
```
Comment (janscience) [inline thunderfish/peakdetection.py]: do we really need the - win_shift_indices/2 ?

File: thunderfish/peakdetection.py [hunk 7]
```diff
@@ -2,6 +2,84 @@
...
+    :param percentile_th: (int). Threshold for peak detection is the given percentile of the amplitude for win_shift wide windows.
+    :param th_factor: (float). The threshold for peak detection is the inter-percentile-range multiplied by this factor.
+
+    :return: threshold: (1-D array). Array with same size as data, including a threshold for each window.
+    """
+    threshold = np.zeros(len(data))
+    win_shift_indices = int(win_shift * samplerate)
+
+    # Make sure percentile_th is correct, regardless of what the user sets as percentile_th
+    if percentile_th < 50.0:
```
Comment (janscience) [inline thunderfish/peakdetection.py]: with the np.abs() below we do not need this any more!

File: thunderfish/peakdetection.py [hunk 8]
```diff
@@ -2,6 +2,84 @@
 import numpy as np
 
 
+def std_threshold(data, samplerate, win_shift, std_factor=5.):
+    """
```
Comment (janscience) [inline thunderfish/peakdetection.py]: as helper functions the *_threshold functions() should go below the detect_peaks() functions (right before the trim() functions)

**Pull request merged**